libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tooling/includeKotlinToolingMetadataInApk.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.tooling
+
+import com.android.build.gradle.AppExtension
+import org.gradle.api.Project
+
+internal fun Project.includeKotlinToolingMetadataInApk() {
```

Comment:
Maybe add a helper method `Project.isAndroidApplication` (based on having `AppExtension`) and then you can reuse existing `Project.forEachVariant` with
```
if (project.isAndroidApplication) project.forEachVariant { it.registerPostJava... }
```

Reply:
I am not sure if this will improve the code. 
Using this approach I see two additional issues over the current implementation. 

1) If executing `Project.isAndroidApplication` here, then we might run into issues regarding about "when is this executed". Maybe the Android plugin will be applied later?

2) Using `Project.forEachVariant` is a cool api, but kind of misses the intention here: 
`forEachVaraint` combines variants provided by applications as well as Libraries, etc. Here, on the other hand, I really just *explicitly* care about applications. So I think it is better to just use the corresponding API here!

Reply:
Keeping it as-is sounds good.